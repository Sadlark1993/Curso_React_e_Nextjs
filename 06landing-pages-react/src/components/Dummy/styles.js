import styled, { css } from 'styled-components';

export const titleStyle = styled.div`
  ${({ theme }) => css``}
`;
