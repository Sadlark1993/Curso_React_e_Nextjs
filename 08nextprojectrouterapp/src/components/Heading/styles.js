import styled, { css } from 'styled-components';

const mediaFont = (theme) => css`
  @media ${theme.media.mobile} {
    font-size: ${theme.fonts.size.xlarge};
  }
`;

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.size.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.size.large};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.size.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.size.xhuge};
    ${mediaFont(theme)};
  `,
};

export const titleStyle = styled.h1`
  ${({ theme, darktitle, size, uppercase }) => css`
    color: ${darktitle ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)}
    text-transform: ${uppercase ? 'uppercase' : 'none'};
  `}
`;
