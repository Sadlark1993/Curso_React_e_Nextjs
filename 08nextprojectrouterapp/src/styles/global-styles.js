import { createGlobalStyle, css } from 'styled-components';
import { theme } from './themes';

export const GlobalStyles = createGlobalStyle`

  ${'' /* reset-start */}

  * {
    margin: 0;
    padding: 0;
    text-decoration: none;
    box-sizing: border-box;
  };

  ul {
    list-style: none;
  }

  ${'' /* reste-end */}


  html {
    font-size: 62.5%;
    scroll-behavior: smooth;
  }

  body {
    font-size: 1.6rem;
    font-family: ${theme.fonts.family.default}
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: ${theme.fonts.family.secondary};
    font-weight: 900;
  }

  p {
    margin: ${({ theme }) => theme.spacing.medium} 0;
  }

  ul, ol{
    margin: ${({ theme }) => theme.spacing.medium} 0;
    padding: ${({ theme }) => theme.spacing.medium} 0;
  }

  a{
    color: ${({ theme }) => theme.colors.sakuraz};
  }
`;
