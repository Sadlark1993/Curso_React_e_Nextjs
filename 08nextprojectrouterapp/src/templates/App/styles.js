import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${(props) => css`
    background: ${props.background};
  `}
  color: ${({ theme }) => theme.colors.boom};
  font-size: 18px;
  &:hover {
    color: black;
  }
  &::after {
    content: '----';
  }
`;
