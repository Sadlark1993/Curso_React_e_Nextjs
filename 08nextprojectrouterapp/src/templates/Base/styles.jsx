import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-top: 5.4rem;

    @media ${theme.media.mobile} {
      padding-top: 0;
    }
  `}
`;
