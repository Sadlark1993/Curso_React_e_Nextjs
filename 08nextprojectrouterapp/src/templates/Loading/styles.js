import styled, { css, keyframes } from 'styled-components';

const rotate = () => keyframes`
  0%{
    transform: translate(-50%, -50%) rotate(0deg);
  }
  100%{
    transform: translate(-50%, -50%) rotate(360deg);
  }
`;

export const Container = styled.div`
  ${({ theme }) => css`
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    background: ${theme.colors.primaryColor};
    z-index: 10;

    &:after,
    &:before {
      content: '';
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      border-width: 0.5rem;
      border-style: solid;
      border-color: transparent;
      border-radius: 50%;
    }

    &:after {
      width: 6rem;
      height: 6rem;
      border-left: 0.5rem solid ${theme.colors.secondaryColor};
      border-top: 0.5rem solid ${theme.colors.secondaryColor};
      animation: ${rotate()} 600ms linear infinite;
    }

    &:before {
      width: 2rem;
      height: 2rem;
      border-left: 0.5rem solid ${theme.colors.secondaryColor};
      border-top: 0.5rem solid ${theme.colors.secondaryColor};
      animation: ${rotate()} 600ms linear reverse infinite;
    }
  `}
`;
