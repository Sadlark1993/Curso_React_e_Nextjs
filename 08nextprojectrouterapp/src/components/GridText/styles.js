import styled, { css } from 'styled-components';
import { titleStyle } from '../Heading/styles';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextContainer} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};
    width: 100%;
    overflow: hidden;
    padding: ${theme.spacing.small};

    @media ${theme.media.mobile} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${titleStyle} {
      position: relative;
      left: 5rem;
    }

    ${titleStyle}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -6rem;
      transform: rotate(12deg);
      font-style: italic;
      font-weight: 900;
    }
  `}
`;
