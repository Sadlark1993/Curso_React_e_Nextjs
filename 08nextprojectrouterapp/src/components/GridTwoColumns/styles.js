import styled, { css } from 'styled-components';
import { titleStyle } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacing.large};

    @media ${theme.media.mobile} {
      grid-template-columns: 1fr;
    }

    ${titleStyle} {
      margin-bottom: ${theme.spacing.xlarge};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.mobile} {
      margin-bottom: ${theme.spacing.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Image = styled.img`
  ${({ theme }) => css``}
`;
