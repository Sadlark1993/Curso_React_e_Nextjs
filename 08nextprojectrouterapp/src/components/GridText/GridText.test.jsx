import { renderTheme } from '../../styles/render-theme';
import { GridText } from '.';
import { screen } from '@testing-library/react';
import mock from './mock';

describe('<GridText />', () => {
  it('Should render', () => {
    const { component } = renderTheme(<GridText {...mock} />);
    expect(screen.getAllByRole('heading')).toHaveLength(4);
    expect(component).toMatchSnapshot();
  });
});
