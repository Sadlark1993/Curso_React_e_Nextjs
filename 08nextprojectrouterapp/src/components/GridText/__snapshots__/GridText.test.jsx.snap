// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridText /> Should render 1`] = `undefined`;
